counterfactual/counterfactual
# Basic lifecycle test for GuessNumber App

Test for happy path of a Counterfactual app:
1. Deploy & fund multisig with A and B as signers
2. Deploy GuessNumber app
3. Deploy StateChannel via Registry
4. `setState` on StateChannel with a terminal state
5. `setResolution` on StateChannel
6. `setNonce` and `finalizeNonce` on NonceRegistry
7. `executeStateChannelConditionalTransfer` on ConditionalTransfer from multisig
8. Confirm that A and B's ETH balances are correct

Misc:
* Helper class for dealing with Multisigs
* New function in Utils: `deployContractViaRegistry`
* Fix security issue in MinimumViableMultisig.setup
* Fix delegatecall bug in `executeStateChannelConditionalTransfer`

TODO: 
- ~Fix tests~
- ~Merge in `develop`~
